spring-rabbit/src/main/java/org/springframework/amqp/rabbit/logback/AmqpAppender.java

```diff
@@ -409,34 +535,84 @@ public void setClientConnectionProperties(String clientConnectionProperties) {
 
 	@Override
 	public void start() {
-		super.start();
-		this.routingKeyLayout.setPattern(this.routingKeyLayout.getPattern()
-				.replaceAll("%property\\{applicationId\\}", this.applicationId));
-		this.routingKeyLayout.setContext(getContext());
-		this.routingKeyLayout.start();
-		this.locationLayout.setContext(getContext());
-		this.locationLayout.start();
-		this.connectionFactory = new CachingConnectionFactory();
-		this.connectionFactory.setHost(this.host);
-		this.connectionFactory.setPort(this.port);
-		if (this.addresses != null) {
-			this.connectionFactory.setAddresses(this.addresses);
+		ConnectionFactory rabbitConnectionFactory = createRabbitConnectionFactory();
+		if (rabbitConnectionFactory != null) {
+			super.start();
```

Comment:
Logback looks good:
```
15:36:01,343 |-ERROR in org.springframework.amqp.rabbit.logback.AmqpAppenderIntegrationTests$EnhancedAppender[AMQP] - Failed to create customized Rabbit ConnectionFactory. java.lang.IllegalStateException: keyStore property required
	at java.lang.IllegalStateException: keyStore property required
	at 	at org.springframework.util.Assert.state(Assert.java:640)
	at 	at org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean.setUpSSL(RabbitConnectionFactoryBean.java:484)
	at 	at org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean.createInstance(RabbitConnectionFactoryBean.java:458)
	at 	at org.springframework.amqp.rabbit.connection.RabbitConnectionFactoryBean.createInstance(RabbitConnectionFactoryBean.java:64)
	at 	at org.springframework.beans.factory.config.AbstractFactoryBean.afterPropertiesSet(AbstractFactoryBean.java:134)
	at 	at org.springframework.amqp.rabbit.logback.AmqpAppender.createRabbitConnectionFactory(AmqpAppender.java:570)
	at 	at org.springframework.amqp.rabbit.logback.AmqpAppender.start(AmqpAppender.java:538)
...
15:36:01,343 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting level of logger [org.springframework.amqp.rabbit.logback] to DEBUG
15:36:01,343 |-INFO in ch.qos.logback.classic.joran.action.LoggerAction - Setting additivity of logger [org.springframework.amqp.rabbit.logback] to false
15:36:01,343 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [AMQP] to Logger[org.springframework.amqp.rabbit.logback]
15:36:01,344 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
15:36:01,344 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
15:36:01,344 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
15:36:01,345 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@4e268090 - Registering current configuration as safe fallback point
```
At least I see analogy with `JMSQueueAppender`:
```
} catch (Exception e) {
      addError("Error while activating options for appender named [" + name
          + "].", e);
    }

    if (this.queueConnection != null && this.queueSession != null
        && this.queueSender != null) {
      super.start();
}
```
and `SMTPAppenderBase`:
```
if (session == null) {
      addError("Failed to obtain javax.mail.Session. Cannot start.");
      return;
    }
```
in the Logback out-of-the-box.